src/staticmaps.js

```diff
@@ -298,62 +301,43 @@ class StaticMaps {
     });
   }
 
-  drawLines() {
-    return new Promise(async (resolve) => {
-      if (!this.lines.length) resolve(true);
+  async drawLines() {
+    if (!this.lines.length) return true;
+    const chunks = chunk(this.lines, LINE_RENDER_CHUNK_SIZE);
```

Comment:
Instead of doing it with chunk, bluebird implements a concurrency function in the promise.map function  http://bluebirdjs.com/docs/api/promise.map.html 

it will be even faster since it will resolve all of them in parallel with a maximum concurrency number, instead of waiting on each chunk.

Reply:
They're actually not promises anymore and there's only 1 composite call that combines them all.